StoreManagementService/tests/PBJ.StoreManagementService.Api.IntegrationTests/ControllerTests/PostControllerTests.cs

```diff
@@ -0,0 +1,274 @@
+        {
+            //Arrange
+            //Act
+            var response = await ExecuteWithStatusCodeAsync(
+                $"{TestingConstants.PostApi}?id={string.Empty}", HttpMethod.Get);
+
+            //Assert
+            response.StatusCode.Should().Be(HttpStatusCode.BadRequest);
+        }
+
+        [Fact]
+        public async Task CreateAsync_WhenRequestModelIsValid_ReturnsOk()
+        {
+            //Arrange
+            var user = await _dataManager.CreateUserAsync();
+
+            var postRequestModel = _fixture.Build<CreatePostRequestModel>()
+                .With(x => x.UserId, user.Id).Create();
+
+            var requestBody = BuildRequestBody(postRequestModel);
+
+            //Act
+            var (postDto, response) = await ExecuteWithFullResponseAsync<PostDto>(
+                TestingConstants.PostApi, HttpMethod.Post, requestBody);
+
+            //Assert
+            response.StatusCode.Should().Be(HttpStatusCode.OK);
+
+            postDto?.UserId.Should().Be(postRequestModel.UserId);
+        }
+
+        [Fact]
+        public async Task CreateAsync_WhenRequestModelIsNotValid_ReturnsBadRequest()
+        {
+            //Arrange
+            var postRequestModel = _fixture.Build<CreatePostRequestModel>()
+                .With(x => x.Content, string.Empty).Create();
+
+            var requestBody = BuildRequestBody(postRequestModel);
+
+
+            //Act
+            var response = await ExecuteWithStatusCodeAsync(
+                TestingConstants.PostApi, HttpMethod.Post, requestBody);
+
+            //Assert
+            response.StatusCode.Should().Be(HttpStatusCode.BadRequest);
+        }
+
+        [Fact]
+        public async Task UpdateAsync_WhenEntityExists_ReturnsOk()
+        {
+            //Arrange
+            var post = await _dataManager.CreatePostAsync();
+
+            var postRequestModel = _fixture.Create<UpdatePostRequestModel>();
+
+            var requestBody = BuildRequestBody(postRequestModel);
+
+            //Act
+            var (postDto, response) = await ExecuteWithFullResponseAsync<PostDto>(
+                $"{TestingConstants.PostApi}?id={post.Id}", HttpMethod.Put, requestBody);
+
+            //Assert
+            response.StatusCode.Should().Be(HttpStatusCode.OK);
+
+            postDto?.UserId.Should().Be(post.UserId);
+        }
+
+        [Fact]
+        public async Task UpdateAsync_WhenEntityNotExists_ReturnsNotFound()
+        {
+            //Arrange
+            var postRequestModel = _fixture.Create<UpdatePostRequestModel>();
+
+            var requestBody = BuildRequestBody(postRequestModel);
+
+            //Act
+            var response = await ExecuteWithStatusCodeAsync(
+                $"{TestingConstants.PostApi}?id={0}", HttpMethod.Put, requestBody);
+
+            //Assert
+            response.StatusCode.Should().Be(HttpStatusCode.NotFound);
+        }
+
+        [Theory]
+        [InlineData(1)]
```

Comment:
There is no need in using this attribute for only one data case. Either add more cases, or move this to constant in the case or outside of it